java/code/src/com/redhat/rhn/domain/entitlement/VirtualizationEntitlement.java

```diff
@@ -63,7 +63,9 @@ public boolean isAllowedOnServer(Server server) {
      * {@inheritDoc}
      */
     public boolean isAllowedOnServer(Server server, ValueMap grains) {
+        String type = grains.getOptionalAsString("virtual").orElse("physical");
+        String subtype = grains.getOptionalAsString("virtual_subtype").orElse("");
         return super.isAllowedOnServer(server) &&
-                grains.getOptionalAsString("virtual").orElse("physical").equals("physical");
+                (type.equals("physical") || (type.equals("xen") && subtype.equals("Xen Dom0")));
```

Comment:
Looks now very similar to: https://github.com/uyuni-project/uyuni/blob/master/java/code/src/com/suse/manager/reactor/hardware/HardwareMapper.java#L536-L543

Not sure if it makes sense to reuse this code. I leave it up to you.

Reply:
right. I refactored the code to share this piece